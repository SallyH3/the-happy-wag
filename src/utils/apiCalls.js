export const getAnimals = async (url, options) => {
  const response = await fetch(url, options) 
  const result = await response.json()
  if(!response.ok) {
    throw Error('Error fetching data')
  }
  return result
}

//tests:
//fetch correct param
//good response
//one for response error
