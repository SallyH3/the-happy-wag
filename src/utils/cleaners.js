export const cleanContactInfo = (contactInfo) => {
  return {
    city: contactInfo.address.city,
    state: contactInfo.address.state,
    email: contactInfo.email
  }
}