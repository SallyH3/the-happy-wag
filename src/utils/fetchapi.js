// import App from '../containers/App';

// export const getAnimals = (url, options) => {
//   return fetch(url)
//   .then(response => response.json())
//   .catch(error => console.log(Error,( 'Error loading animals')))
// }