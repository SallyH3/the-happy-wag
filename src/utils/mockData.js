export const mockAnimals = [
  {
    age: "Baby",
    contact: {email: "lastpawtotherescue@yahoo.com"},
    description: "Sneak Peak!! Hey, you all!! Ducky here. I will be arriving to Wisconsin on Sunday from Tx. I am a...",
    gender: "Female",
    id: 44853384,
    name: "Ducky",
    photos: (3) [{}, {}, {}],
    species: "Dog"
  },
  {
    age: "Baby",
    contact: {email: "furrfoundation@gmail.com"},
    description: "Marcel is such a lovable little boy. He loves to have his...",
    gender: "Male",
    id: 44853380,
    name: "Marcel",
    photos: [{}],
    species: "Cat"
  }
]

export const mockSelectedAnimal =  {
  age: "Baby",
  contact: {email: "furrfoundation@gmail.com"},
  description: "Marcel is such a lovable little boy. He loves to have his...",
  gender: "Male",
  id: 44853380,
  name: "Marcel",
  photos: [{}],
  species: "Cat"
}

export const mockPhotos = [
  {
    medium: "https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/44868950/1/?bust=1559597198&width=300"
  }
]