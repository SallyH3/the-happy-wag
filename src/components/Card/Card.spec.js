import React from 'react';
import { Card } from './index';
import { shallow } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import { mockPhotos } from '../../utils/mockData';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card photo={mockPhotos}/>
    )
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})