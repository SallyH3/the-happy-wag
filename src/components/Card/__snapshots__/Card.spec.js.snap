// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/CardDetails/undefined"
>
  <section
    className="card-display"
  >
    <article
      className="card-layout"
    >
      <img
        src="https://dl5zpyw5k3jeb.cloudfront.net/photos/pets/44868950/1/?bust=1559597198&width=300"
      />
      <p
        className="details name"
      />
      <p
        className="details"
      >
        Age: 
      </p>
      <p
        className="details"
      >
        Gender: 
      </p>
      <p
        className="details"
      >
        Species: 
      </p>
      <p
        className="card-back"
      />
    </article>
  </section>
</Link>
`;
